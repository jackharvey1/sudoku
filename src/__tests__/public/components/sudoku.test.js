import React from 'react';
import Sudoku from '../../../public/components/sudoku';
import ShallowRenderer from 'react-test-renderer/shallow';
import { emptySudoku } from '../../sudokus.json';

it('renders the Sudoku component as intended', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<Sudoku
        grid={emptySudoku}
        lockedCells={[]}
        selectSquare={() => {}}
        selectedBox={0}
        selectedSquare={0}
        isUnderCheck={false}
        solution={emptySudoku}
    />);
    const output = renderer.getRenderOutput();
    expect(output).toMatchSnapshot();
});
