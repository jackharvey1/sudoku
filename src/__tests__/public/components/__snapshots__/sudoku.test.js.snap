// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Sudoku component as intended 1`] = `
<div>
  <div
    className="css-1mzzuk6"
  >
    <Box
      box={0}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={1}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={2}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </div>
  <div
    className="css-1mzzuk6"
  >
    <Box
      box={3}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={4}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={5}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </div>
  <div
    className="css-1mzzuk6"
  >
    <Box
      box={6}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={7}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
    <Box
      box={8}
      lockedCells={Array []}
      onClick={[Function]}
      selectedBox={0}
      selectedSquare={0}
      values={
        Array [
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
          "",
        ]
      }
    />
  </div>
</div>
`;
